presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/ReplaceWindowWithRowNumber.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.matching.Captures;
+import io.prestosql.matching.Pattern;
+import io.prestosql.metadata.FunctionId;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.sql.planner.PlanNodeIdAllocator;
+import io.prestosql.sql.planner.iterative.Rule;
+import io.prestosql.sql.planner.plan.PlanNode;
+import io.prestosql.sql.planner.plan.RowNumberNode;
+import io.prestosql.sql.planner.plan.WindowNode;
+import io.prestosql.sql.tree.QualifiedName;
+
+import java.util.Optional;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.prestosql.sql.planner.plan.Patterns.window;
+
+public class ReplaceWindowWithRowNumber
+        implements Rule<WindowNode>
+{
+    private final Pattern<WindowNode> pattern;
```

Comment:
Unlike the other usual cases, it's not the static variable. Do we still need to capitalize it?

Reply:
That's the convention we follow throughout most of the rules. The same with the captured `CHILD`. I think I saw it capitalized in another of the rules you add.

Reply:
@kasiafi, the difference here is that the pattern is an instance field because it needs to be initialized upon construction due to the dependency on Metadata. We only capitalize static fields (per common Java style)

Reply:
ah, right.